src/style_builder.cr

```diff
@@ -17,6 +17,28 @@ module Mint
     end
   end
 
+  class StylePool
+    @pool = NamePool(Ast::Style, Nil).new
+    @cache = {} of Ast::Style => String
+
+    def initialize(@optimize : Bool = false)
+    end
+
+    def of(subject : Ast::Style, id : String? = nil)
+      if @optimize
+        @pool.of(subject, nil)
+      else
+        @cache[subject] ||= begin
+          if id
+            "#{id}_#{subject.name.value.gsub('-', '_')}"
+          else
+            subject.name.value
```

Comment:
Shouldn't this be escaped too?

```suggestion
            subject.name.value.gsub('-', '_')
```

Reply:
I don't think so, why?

Reply:
In the same vein as the line above I guess.

Reply:
Oh right, I was thinking you commented on *that* line :joy: